### ISSUE TITLE
"No method" error happens when instance variable is set in initialize method

### ISSUE THREAD

**Issue by kishima** [mrubyc/mrubyc#50]
## Precondition

- OS Windows10/cygwin64
- mruby 1.3.0
- mruby/c 
commit 7549788e98bf1fe29d0e1ddaa9f3ccd5f4091cd9 (HEAD -> master, origin/master, origin/HEAD)
Author: Hirohito Higashi <uzuki.local>
Date:   Wed May 30 17:27:00 2018 +0900

## Procedure to reproduce

to run the following byte-compiled script test.rb by mrbc.

```rb
class Test
        def initialize
                puts "initilize"
                @val="test"
        end
        def method1
                puts "method1"
                puts @val
        end

end

t=Test.new
t.method1
```

## Expected result

     $ ./mrubyc.exe test.mrb
    initilize 
    method1
    test

## Observed result

     $ ./mrubyc.exe test.mrb
    initilize 
    No method. vtype=23 method='method1'

## Comment

Is that expected behavior? I also saw the same error on ubuntu14.04 32bit.
I'm trying to find cause for my study...

**Comment by kishima** [comment]
I have tested other cases. I saw same error when I create 2 instances which have an instance variable. It looks like simply instance variable is not supported yet.
If it's correct, I'll close this issue.

**Comment by kaz0505** [comment]
This bug will be made from my added code, which is related to block execution.
I'm sorry...

I'll check and fix it.

**Comment by kishima** [comment]
Don’t say sorry. Thank you for your confirmation!

**Comment by kishima** [comment]
After my study, I found that "new" method doesn't return a new object in this case. It works if I put "self" at the end of "initialize" method. This could be a workaround.
I think this behavior comes from c_object_new(). "new" method returns the result of "initialize" method directory.
I think it is needed to run a VM in c_object_new() like a method call with a block. I have implemented my idea and it works but it's not tested well. Once I get a confidence, I'd like to send a pull request if this is still open at the time.
